src/history.h

```diff
@@ -40,9 +40,13 @@ struct io_streams_t;
 typedef std::vector<wcstring> path_list_t;
 
 enum history_search_type_t {
-    // The history searches for strings containing the given string.
+    // We don't know what type of search to perform.
+    HISTORY_SEARCH_TYPE_UNKNOWN,
```

Comment:
Instead of introducing this sentinel "unknown" search type, I would rather see a  "has_history_search_type" boolean or something in builtin_history.